app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def require_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
You do not cache the result of User.find. During a request `current_user` can and is usually called heavily and each time you would hit your server with a query.

HOMEWORK: watch this railscast and fix the implementation - http://railscasts.com/episodes/1-caching-with-instance-variables